Detectors/FIT/common/simulation/src/Digitizer.cxx

```diff
@@ -24,15 +25,25 @@ using namespace o2::fit;
 
 ClassImp(Digitizer);
 
-void Digitizer::process(const std::vector<o2::fit::HitType>* hits, Digit* digit)
-
+double get_time(const std::vector<double>& times, double signal_width)
 {
-  //parameters constants TO DO: move to class
+  TH1F hist("time_histogram", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
+  auto signalForm = [](double x) {
+    return -(exp(-0.83344945 * x) - exp(-0.45458 * x));
+  };
+  for (auto time : times)
+    for (int bin = hist.FindBin(time); bin < hist.GetSize(); ++bin)
+      if (hist.GetBinCenter(bin) > time)
+        hist.AddBinContent(bin, signalForm(hist.GetBinCenter(bin) - time));
+  double maximum{ hist.GetMaximum() };
+  int binfound{ hist.FindFirstBinAbove(maximum * 0.2) };
+  //  std::cout<<"@@@@@!!! max "<<maximum<<" binfound "<<binfound<<" return "<<hist.GetBinCenter(binfound)<<std::endl;
+  return hist.GetBinCenter(binfound);
+}
 
-  constexpr Float_t C_side_cable_cmps = 2.877; //ns
-  constexpr Float_t A_side_cable_cmps = 11.08; //ns
-  constexpr Float_t signal_width = 5.;         // time gate for signal, ns
+void Digitizer::process(const std::vector<o2::fit::HitType>* hits, Digit* digit, std::vector<std::vector<double>>& channel_times)
```

Comment:
Do you really need vector of vectors? It seems in the process method you always set the outer vector to the same size (mMCP). If this is fixed, better use array<vectot<double>,size).

Reply:
Hi @AllaMaevskaya 

I think the best would be to unstage your changes (``git reset --soft HEAD~7``), then recover all files which should not be affected by this PR (``git checkout -- Common/Field/include/Field/MagneticWrapperChebyshev.h`` etc.), then rebase to fresh dev, solve conflicts if any and push again.

Cheers,
 Ruben

Reply:
Hello
mMCP=208 is the number of channels for T0 but it can be different for V0. So better to keep it as it is before we will decide how to call it for both detectors

Best regards
Alla

Reply:
@AllaMaevskaya ok, I still did not understand if the Digitizer will be serving only T0 or all subdetectors.